addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -1439,6 +1439,51 @@ export function makeActionManager(env, router = _router) {
             }
             throw new InvalidButtonParamsError("Missing type for doActionButton request");
         }
+        if (action.context?.load_first_embedded_action && action.embedded_action_ids) {
+            const embeddedActionsOrder = JSON.parse(
+                browser.localStorage.getItem(
+                    `orderEmbedded${action.id}+${params.resId || ""}+${user.userId}`
+                )
+            );
+            const embeddedActionId = embeddedActionsOrder?.at(0);
+            const embeddedAction = action.embedded_action_ids?.find(
+                (embeddedAction) => embeddedAction.id === embeddedActionId
+            );
+            if (embeddedAction) {
+                const embeddedActions = [
+                    ...action.embedded_action_ids,
+                    {
+                        id: false,
+                        name: action.name,
+                        parent_action_id: action.id,
+                        parent_res_model: action.res_model,
+                        action_id: action.id,
+                        user_id: false,
+                        context: {},
+                    },
+                ];
+                const context = {
+                    ...action.context,
+                    active_id: params.resId,
+                    active_model: params.resModel,
+                    current_embedded_action_id: embeddedActionId,
+                    parent_action_embedded_actions: embeddedActions,
+                    parent_action_id: action.id,
+                };
+                await doActionButton({
```

Comment:
we could add a `executeEmbeddedAction` in the action service API, which would allow to factorize some code from the control panel and here. If you want, @pparidans can do it as he's currently refactoring the embedded action logic.

Reply:
@dise-odoo Don't worry too much about factorizing it, I'll do it in the refactoring :wink: